### ISSUE TITLE
SASS not working with multiple independent SCSS files. Causes hanging on front-end

### ISSUE THREAD

**Issue by x0rsw1tch** [dotCMS/core#18301]
> Reproducible on 5.2.8 demo site.

### Describe the bug

When creating multiple independent SCSS files that are referenced/compiled separately using built in compiler:
- The front-end hangs on the SASS files
- The .dotsass referenced files result in an HTTP 504 error.
- Page doesn't completely load correctly due to missing CSS files.

Screenshot of demo site after reproducing.

![image](https://user-images.githubusercontent.com/9543257/78940010-28adba80-7a83-11ea-8dcf-03d0a260bc7f.png)

### Steps to reproduce the behavior:
1. Create a new test SCSS file with the following code, nothing fancy or complicated:

```
.row.no-gutter {
	margin-left: 0;
	margin-right: 0;

	div[class*='col-'] {
		padding-left: 0;
		padding-right: 0;
	}
}
```

2. Upload file to `/application/themes/travel/css`
2.5 Change the file extension to `.scss` at fields: `File Name` and `Title`
3. Edit  `/application/themes/travel/html_head.vtl` to reference new file

```
<link rel="stylesheet" href="${dotTheme.path}css/test.dotsass">
```

4. Save/Publish `html_head.vtl`
5. Everything should still be working at this point, including the new SCSS file.
6. Open `/application/themes/travel/css/styles.scss` for editing, lock the file.
7. Do not change contents of file, just Save/Publish.
8. Try to load `https://demo.dotcms.com/store/`, page will hang. The `.dotsass` referenced files will timeout with a 504 error. Once this happens the page will display, but without any styling.

### Expected behavior

Each `.dotsass` referenced file compiles independently of each other and everything is peachy king.

### Additional context

I have run into this issue on multiple clients' sites. If we want to use the built-in SASS compiler, we're stuck with one master SCSS file. This is usually not a problem, but there are instances where having independent SCSS files with specific targets in mind.

**Comment by john-thomas-dotcms** [comment]
Reported by customer: https://dotcms.zendesk.com/agent/tickets/104136

**Comment by bryanboza** [comment]
Fixed, tested on release-21.09 // Docker
